tests/test_bs.py

```diff
@@ -150,22 +152,12 @@ def setup_server(deployment_name, deployment_file, namespace):
 @pytest.fixture(scope='module')
 def setup_poet(request):
     poet_deployment_name = 'poet'
-
-    def fin():
-        delete_deployment(poet_deployment_name, testconfig['namespace'])
-
-    request.addfinalizer(fin)
```

Comment:
I don't think that removing all the finalisers is a good practice.
Each fixtures define the setup and the teardown of itself.
I understand that teardown takes time. This is because we wait for each deployment to be deleted (blocked on delete)
Instead we can add additional parameter to "delete_deployment" - block=True\False 
Thus we will call "wait_to_deployment_to_be_deleted" only if block=True

This is more constructed then rely on the finalizer of "set_namespace" to teardown all.

Reply:
Why do this if we're going to use block=False all the time ?
wouldn't it be a better practice to add code only when it's needed? when the use case for blocking deletion will raise we can add this

Reply:
can do this though as it just as adding a line instead of removing. let me know what you think

Reply:
When people write new tests, and use the fixtures in their tests, it will be more readable to understand what the fixture does by reading only it's code. (They don't need to understand that there are other fixtures which are called during teardown that handle this)

Reply:
To my opinion this is better, even though it is generally better practice to add code when you need it

Reply:
I get it, only if when `delete` is False I don’t even send the api call to delete. sending an api call for each of the deployments will slow us down without reason as deleting the namespace does the job we need in a shorter time

Reply:
#940